cli/src/commands/diffedit.rs

```diff
@@ -119,13 +122,17 @@ don't make any changes, then the operation will be aborted.",
             .write()?;
         // rebase_descendants early; otherwise `new_commit` would always have
         // a conflicted change id at this point.
-        let num_rebased = tx.repo_mut().rebase_descendants(command.settings())?;
+        let num_rewritten = if args.restore_descendants {
+            tx.repo_mut().reparent_descendants(command.settings())?
+        } else {
+            tx.repo_mut().rebase_descendants(command.settings())?
+        };
         if let Some(mut formatter) = ui.status_formatter() {
             write!(formatter, "Created ")?;
             tx.write_commit_summary(formatter.as_mut(), &new_commit)?;
             writeln!(formatter)?;
-            if num_rebased > 0 {
-                writeln!(formatter, "Rebased {num_rebased} descendant commits")?;
+            if num_rewritten > 0 {
+                writeln!(formatter, "Rewritten {num_rewritten} descendant commits")?;
```

Comment:
nit: Rewritten -> Rewrote

Also - and I know we talked about this on Discord, but I didn't think about it until now - I'm worried that users who don't pass `--restore-descendants` will be confused that this command says "Rewrote 5 descendant commits" when all other commands say "Rebased 5 descendant commits". I'm leaning towards  using "Rebased" for both cases or maybe "Rebased 5 descendant commits" and "Rebased 5 descendant commits (while preserving their content)" or something like that.

Reply:
I've added your proposed phrasing. If we add this option to more commands, it will make sense to also factor out the message printing in order not to repeat the same text (and code) everywhere.

Reply:
@martinvonz What about using "Rebased 5 descendant commits (no content change)" instead of "Rebased 5 descendant commits (while preserving their content)"?

Reply:
I think that sounds a bit more like something that happened to be the case given some circumstance than something that's guaranteed to be the case. Perhaps the user will think that it was something about those commits that made it so there was no content change. Does that make sense?

Reply:
It does! Leaving as-is.